receiver/bigipreceiver/scraper.go

```diff
@@ -96,21 +96,23 @@ func (s *bigipScraper) scrape(ctx context.Context) (pmetric.Metrics, error) {
 		}
 	}
 
-	// scrape metrics for pool members
-	poolMembers, err := s.client.GetPoolMembers(ctx, pools)
-	if errors.Is(err, errCollectedNoPoolMembers) {
-		scrapeErrors.AddPartial(1, err)
-		s.logger.Warn("Failed to scrape pool member metrics", zap.Error(err))
-	} else {
-		if err != nil {
-			scrapeErrors.AddPartial(1, err)
-			s.logger.Warn("Failed to scrape some pool member metrics", zap.Error(err))
-		}
-
-		collectedMetrics = true
-		for key := range poolMembers.Entries {
-			poolMemberStats := poolMembers.Entries[key]
-			s.collectPoolMembers(&poolMemberStats, now)
+	if pools != nil {
```

Comment:
I went with this approach as I figured we still wanted to try GetNodes later on and pass back any metrics we got from it (and/or GetVirtualServers).  Of course we could shuffle GetNodes up above GetPoolMembers and have the GetPoolMembers as the last of the "scrape metrics" blocks and have at the end of the scrape function something like
```
if pools == nil {
	if !collectedMetrics {
		return pmetric.NewMetrics(), errScrapedNoMetrics
	}
        return s.mb.Emit(), scrapeErrors.Combine()
}

[ the current getpoolmembers code]

// still need this one in case GetPoolMembers failed
if !collectedMetrics {
        return pmetric.NewMetrics(), errScrapedNoMetrics
}
return s.mb.Emit(), scrapeErrors.Combine()
```
but I need to have the `!collectedMetrics` check twice and still have the a level of nesting as we have two possible return results.
If this is the preference let me know and I'll update the PR, thanks.

Reply:
Alright, thank you. I think it's fine. We can always refactor later if necessary.